sw/device/tests/sram_ctrl_sleep_sram_ret_contents_test.c

```diff
@@ -0,0 +1,116 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_aon_timer.h"
+#include "sw/device/lib/dif/dif_pwrmgr.h"
+#include "sw/device/lib/dif/dif_rstmgr.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/aon_timer_testutils.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/pwrmgr_testutils.h"
+#include "sw/device/lib/testing/rstmgr_testutils.h"
+#include "sw/device/lib/testing/test_framework/ottf.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+#define NUM_TEST_WORDS 16
+
+const test_config_t kTestConfig;
+
+// Random data to read/write to/from retention SRAM.
+const uint32_t kTestData[NUM_TEST_WORDS] = {
+    0xe647e5d5, 0x4b5fe6f6, 0x1608a98a, 0x5e347116, 0xb2dc5e92, 0x899e3c0f,
+    0xc98295c2, 0x0fa84434, 0x15747561, 0xfecb5aa1, 0x7a78bb8c, 0x8f9c5d0f,
+    0x49338fbd, 0x093e82cb, 0xaaa58121, 0x5b806f96,
+};
+
+static void retention_sram_check(bool do_write) {
+  mmio_region_t sram_region_ret_base_addr =
+      mmio_region_from_addr(TOP_EARLGREY_SRAM_CTRL_RET_AON_RAM_BASE_ADDR);
+  for (int i = 0; i < NUM_TEST_WORDS; ++i) {
+    if (do_write)
+      mmio_region_write32(sram_region_ret_base_addr, i * sizeof(uint32_t),
```

Comment:
nit: While this is theoretically allowed by [our style guide](https://google.github.io/styleguide/cppguide.html#Conditionals), I find it more readable to  have the curly braces.